rplugin/python3/gdb/scm.py

```diff
@@ -49,6 +50,9 @@ def queryB(self, match):
     def _search(self):
         # If there is a matcher matching the line, call its handler.
         for matcher, func in self.state:
+            # Filter ANSI Escape Sequence
+            reg = r'(\x9B|\x1B\[)[0-?]*[ -\/]*[@-~]'
```

Comment:
I believe there could be a better place for filtering out. The `self.buffer` is an accumulation buffer, and thus, your substitution will be doing repetitive searches in it. It'd be better to hook up before the method `_search` is called. Somewhere closer to the data source.

Reply:
On the other hand, the stream doesn't guarantee that an escape sequence is received atomically. So I recommend moving the substitution just of the matching loop.
Does it substitute all the occurrences?

Reply:
> Does it substitute all the occurrences?

Found the regex [here](https://stackoverflow.com/questions/14693701/how-can-i-remove-the-ansi-escape-sequences-from-a-string-in-python)